### PR TITLE
UCP/PROTO: Adjust performace estimations

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/ucp/rndv/rndv_rkey_ptr.c [hunk 1]
```diff
@@ -23,6 +23,15 @@ typedef struct {
 } ucp_proto_rndv_rkey_ptr_priv_t;
 
 
+static double ucp_proto_rndv_rkey_ptr_overhead() {
```
Comment (yosefe) [inline src/ucp/rndv/rndv_rkey_ptr.c:26]: code style - { on next line

File: src/uct/base/uct_iface.h [hunk 2]
```diff
@@ -205,8 +209,16 @@ enum {
...
+#define UCT_EP_OP_IS_BCOPY(_op) \
+    (UCS_BIT(_op) & (UCS_BIT(UCT_EP_OP_AM_BCOPY) | \
+                     UCS_BIT(UCT_EP_OP_PUT_BCOPY) | \
+                     UCS_BIT(UCT_EP_OP_GET_BCOPY)))
+
+
+#define UCT_EP_OP_IS_ZCOPY(_op) \
+    (UCS_BIT(_op) & (UCS_BIT(UCT_EP_OP_AM_ZCOPY) | \
+                     UCS_BIT(UCT_EP_OP_PUT_ZCOPY) | \
+                     UCS_BIT(UCT_EP_OP_GET_ZCOPY)))
```
Comment (yosefe) [inline src/uct/base/uct_iface.h:221]: make it static inline

File: src/uct/ib/base/ib_iface.c [hunk 3]
```diff
@@ -1705,52 +1705,45 @@ ucs_status_t uct_ib_iface_query(uct_ib_iface_t *iface, size_t xport_hdr_len,
...
         send_post_overhead_bcopy = send_post_overhead_zcopy = 0;
     }
 
-    op = UCT_ATTR_VALUE(PERF, perf_attr, operation, OPERATION, UCT_EP_OP_LAST);
-
     if (perf_attr->field_mask & UCT_PERF_ATTR_FIELD_SEND_PRE_OVERHEAD) {
-        switch (op) {
-        case UCT_EP_OP_AM_BCOPY:
-        case UCT_EP_OP_AM_ZCOPY:
+        if (UCT_EP_OP_IS_BCOPY(op) || UCT_EP_OP_IS_ZCOPY(op)) {
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:1733]: why need this if ?

File: src/uct/ib/base/ib_iface.c [hunk 4]
```diff
@@ -1705,52 +1705,45 @@ ucs_status_t uct_ib_iface_query(uct_ib_iface_t *iface, size_t xport_hdr_len,
...
     }
 
     if (perf_attr->field_mask & UCT_PERF_ATTR_FIELD_SEND_POST_OVERHEAD) {
-        switch (op) {
-        case UCT_EP_OP_AM_BCOPY:
+        if (UCT_EP_OP_IS_BCOPY(op)) {
             perf_attr->send_post_overhead = send_post_overhead_bcopy;
-            break;
-        case UCT_EP_OP_AM_ZCOPY:
+        } else if (UCT_EP_OP_IS_ZCOPY(op)) {
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c:1743]: if ZCOPY {
  send_post_overhead_zcopy
} else {
  send_post_overhead_zcopy
}

File: src/uct/sm/mm/base/mm_iface.c [hunk 5]
```diff
@@ -483,26 +483,38 @@ static ucs_status_t
...
     if (perf_attr->field_mask & UCT_PERF_ATTR_FIELD_SEND_PRE_OVERHEAD) {
         switch (op) {
+        case UCT_EP_OP_AM_SHORT:
+            perf_attr->send_pre_overhead = short_overhead;
+            break;
         case UCT_EP_OP_AM_BCOPY:
-            perf_attr->send_pre_overhead = overhead;
+            perf_attr->send_pre_overhead = am_overhead;
+            break;
+        case UCT_EP_OP_PUT_BCOPY:
```
Comment (yosefe) [inline src/uct/sm/mm/base/mm_iface.c:516]: maybe make PUT_BCOPY fallback to default case as well?

File: src/uct/sm/mm/base/mm_iface.c [hunk 6]
```diff
@@ -512,11 +524,17 @@ uct_mm_estimate_perf(uct_iface_h tl_iface, uct_perf_attr_t *perf_attr)
...
         switch (op) {
+        case UCT_EP_OP_AM_SHORT:
+            perf_attr->recv_overhead = short_overhead;
+            break;
         case UCT_EP_OP_AM_BCOPY:
-            perf_attr->send_pre_overhead = overhead;
+            perf_attr->recv_overhead = am_overhead;
+            break;
+        case UCT_EP_OP_PUT_BCOPY:
+            perf_attr->recv_overhead = put_overhead;
```
Comment (yosefe) [inline src/uct/sm/mm/base/mm_iface.c:534]: should be 0

File: src/uct/ib/base/ib_iface.c (hunk 3 above)
Comment (Artemy-Mellanox) [reply src/uct/ib/base/ib_iface.c]: what if its short?

Comment (yosefe) [reply src/uct/ib/base/ib_iface.c]: right, ok

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**